### PR TITLE
os/bluestore: get rid off statfs update on each txc - fix check

### PR DESCRIPTION
**Pull request by aclamk**
This PR is a temporary to check for proposed solution for a bug caused by #46036

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/os/bluestore/BlueStore.cc [hunk 1]
```diff
@@ -6546,7 +6551,15 @@ int BlueStore::open_db_environment(KeyValueDB **pdb, bool to_repair)
 
 int BlueStore::close_db_environment()
 {
-  _close_db_and_around();
+  // verify that the db is actually closed
+  if (db) {
+    RocksDBStore* rocks_db = dynamic_cast<RocksDBStore*>(db);
+    ceph_assert(rocks_db);
+    ceph_assert(!rocks_db->is_open());
+    delete(db);
```
Comment (ifed01) [inline src/os/bluestore/BlueStore.cc:6559]: IMO 
delete db;
is a better (more common) format..

File: src/os/bluestore/BlueStore.cc [hunk 2]
```diff
@@ -6546,7 +6551,15 @@ int BlueStore::open_db_environment(KeyValueDB **pdb, bool to_repair)
 
 int BlueStore::close_db_environment()
 {
-  _close_db_and_around();
+  // verify that the db is actually closed
+  if (db) {
+    RocksDBStore* rocks_db = dynamic_cast<RocksDBStore*>(db);
```
Comment (ifed01) [inline src/os/bluestore/BlueStore.cc:6556]: if you really want to call is_open() in production - which is questionable as it's called exclusively  from ceph_assert - it worth to make is_open() virtual and do not use dynamic_cast

**Review by ifed01 — COMMENTED** [review]
I think it makes sense to add this new commit to my original PR once we have a final version... @aclamk - are you fine with that?

File: src/os/bluestore/BlueStore.cc (hunk 2 above)
Comment (aclamk) [reply src/os/bluestore/BlueStore.cc:6556]: How about skipping the check altogether?

File: src/os/bluestore/BlueStore.cc (hunk 1 above)
Comment (aclamk) [reply src/os/bluestore/BlueStore.cc:6559]: > I think it makes sense to add this new commit to my original PR once we have a final version... @aclamk - are you fine with that?
Please add it. Without your code this change does not make much sense.

File: src/os/bluestore/BlueStore.cc (hunk 2 above)
Comment (ifed01) [reply src/os/bluestore/BlueStore.cc]: sounds good

File: src/os/bluestore/BlueStore.cc (hunk 1 above)
Comment (ifed01) [reply src/os/bluestore/BlueStore.cc]: done, this one can be closed

**Pull request closed without merging**